src/seedu/addressbook/Main.java

```diff
@@ -105,15 +111,18 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException, RuntimeException  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
-        } catch (Exception e) {
+        }   catch (StorageOperationException e) {
+                throw e;
+
+        }   catch (Exception e) {
             ui.showToUser(e.getMessage());
-            throw new RuntimeException(e);
+                throw new RuntimeException(e);
```

Comment:
Same as above, maybe you can review the code flow and examine if we need to explicitly catch and throw. What happens if you don't catch it here? Also, what is a good usecase for throw?